lib/src/quantized.nr

```diff
@@ -40,27 +40,21 @@ impl Quantized {
 
     fn mul(self: Self, other: Self) -> Self {
         let mut temp: Field = self.x * other.x;
+        let negative = is_negative(temp);
+
+        temp = negative
+            * (
+                21888242871839275222246405745257275088548364400416034343698204186575808495616 - temp
+                    + 1
+                    - temp
+            )
+            + temp;
         let mut bytes: [u8; 32] = [0; 32];
         unsafe {
             bytes = get_bytes(temp);
         }
-
         assert(Field::from_be_bytes::<32>(bytes) == temp);
 
-        let negative = is_negative(temp);
-
-        // To correctly scale down we temporary flip the sign
-        if negative == 1 {
-            temp = 21888242871839275222246405745257275088548364400416034343698204186575808495616
-                - temp
-                + 1;
-            unsafe {
```

Comment:
this chunk from line 57-61 is missing, and I think this has impact both on the gatecount as well as the functionality. Functionality: when the number is negative, we flip the switch and consequently, the bytes get reversed. This is needed for the easy division by 2^16 to work (by chopping off 2 bytes). Additionally, we do the assert because `get_bytes` is an unconstrained function. 

Did you run the tests, apart from the gatecount? To verify the computation is still correct? Because I would think the test fail now. Or am I missing something?

Reply:
Got it. We did the byte decomposition double before but that was not the smart way. So that is what accounts for the gatecount improvement in Noir!